src/libraries/System.Runtime/tests/System/Uri.MethodsTests.cs

```diff
@@ -389,15 +389,17 @@ public static IEnumerable<object[]> Equals_TestData()
         public void Equals(Uri uri1, object obj, bool expected)
         {
             Uri uri2 = obj as Uri;
+
             if (uri1 != null)
             {
                 Assert.Equal(expected, uri1.Equals(obj));
+
                 if (uri2 != null)
                 {
-                    bool onlyCaseDifference = string.Equals(uri1.OriginalString, uri2.OriginalString, StringComparison.OrdinalIgnoreCase);
-                    Assert.Equal(expected || onlyCaseDifference, uri1.GetHashCode().Equals(uri2.GetHashCode()));
+                    Assert.Equal(expected, uri1.GetHashCode() == uri2.GetHashCode());
```

Comment:
This assertion is incorrect. It should really be something along these lines:

```cs
if (expected)
{
    Assert.Equal(uri1.GetHashCode(), uri2.GetHashCode());
}

// contrapositive to above assertion
if (uri1.GetHashCode() != uri2.GetHashCode())
{
    Assert.False(expected);
}
```

Reply:
Changed